sampletester/inputs.py

```diff
@@ -77,6 +74,9 @@ def log_files(indexed_files):
     logging.info('testplan files:\n  {}'.format('\n  '.join(testplan_paths)))
     return indexed_files
 
+  if not file_patterns:
```

Comment:
This looks a bit weird.

Reply:
Not necessary in any way, but I thought it looks kind of weird to define an inner function in the middle of the other code—it breaks the flow as you're parsing the top level, then you have this inner function to parse, and then when it's done you come back to the upper level logic. Just organization for human readers.....